paderbox/io/atomic.py

```diff
@@ -115,31 +120,41 @@ def open_atomic(file, mode, *args, force=False, **kwargs):
 
     >>> with open_atomic(file, 'w') as f:
     ...     print('Name:', f.name)  # doctest: +ELLIPSIS
-    Name: .../tmp.io.txt...
+    Name: ...tmp.io.txt...
     """
     file = normalize_path(file, as_str=True, allow_fd=False)
 
     assert 'w' in mode, mode
 
-    with tempfile.NamedTemporaryFile(
-            mode, *args, **kwargs, prefix=file, dir=os.getcwd()
-    ) as tmp_f:
-        def cleanup():
-            tmp_f.flush()
-            os.fsync(tmp_f.fileno())
-            # os.rename(tmp_f.name, file)  # fails if dst exists
-            os.replace(tmp_f.name, file)
-
-            # Disable NamedTemporaryFile.close(), because the file was renamed.
-            tmp_f.delete = False
-            tmp_f._closer.delete = False
-        try:
-            yield tmp_f
-            if not force:
-                cleanup()
-        finally:
-            if force:
-                cleanup()
+    try:
+        clean = False
+        fname = None
+        with tempfile.NamedTemporaryFile(
+                mode, *args, **kwargs, delete=False, prefix=file, dir=os.getcwd()
+        ) as tmp_f:
+            fname = tmp_f.name
+
+            def cleanup():
+                tmp_f.flush()
+                os.fsync(tmp_f.fileno())
+
+            try:
+                yield tmp_f
+                if not force:
+                    clean = True
+                    cleanup()
+            finally:
+                if force:
+                    clean = True
+                    cleanup()
+    finally:
+        if clean:
```

Comment:
Can you use an if (`if os.name == 'nt'`) statement to minimize the system calls? On network file systems, this is unnecessarily expensive.